src/NHibernate/Action/BulkOperationCleanupAction.cs

```diff
@@ -52,17 +68,16 @@ public BulkOperationCleanupAction(ISessionImplementor session, ISet<string> quer
 			//from H3.2 TODO: cache the autodetected information and pass it in instead.
 			this.session = session;
 
-			ISet<string> tmpSpaces = new HashSet<string>(querySpaces);
-			ISessionFactoryImplementor factory = session.Factory;
-			IDictionary<string, IClassMetadata> acmd = factory.GetAllClassMetadata();
-			foreach (KeyValuePair<string, IClassMetadata> entry in acmd)
-			{
-				string entityName = entry.Key;
-				IEntityPersister persister = factory.GetEntityPersister(entityName);
-				string[] entitySpaces = persister.QuerySpaces;
+			ISet<string> affectedSpaces = new HashSet<string>(querySpaces);
+			ISet<string> affectedQueryCacheSpaces = new HashSet<string>();
 
-				if (AffectedEntity(querySpaces, entitySpaces))
+			var entityPersistersSpaces = session.Factory.GetEntityPersistersSpaces();
+			foreach (var querySpace in querySpaces)
```

Comment:
Why NHibernate.Engine.Query.Sql.NativeSQLQuerySpecification.QuerySpaces is empty, I could not find out @maca88 
NHibernate.Test.BulkManipulation.NativeSQLBulkOperationsWithCache.SimpleNativeSQLInsert_DoesNotEvictEntireCacheWhenQuerySpacesAreAdded test fails

Reply:
Ok but if you look at test case it checks only one space is cleared not all spaces? Why native sql does not detects querySpaces, is it not an issue?

Reply:
in that test case there is only one space, which is `VEHICLE`.

> Why native sql does not detects querySpaces, is it not an issue?

No it is not. It is almost impossible to detect query spaces in a custom sql query as the query may use stored procedures, table functions, views and other dialect specific objects for which we are not able to detect their query spaces.

Reply:
ok